org.lflang/src/org/lflang/generator/c/CCoreFilesUtils.java

```diff
@@ -112,13 +115,16 @@ private static List<String> getThreadSupportFiles(
         SchedulerOption scheduler
     ) {
         return threading ?
-                List.of(
+            Stream.concat(
+                Stream.of(
                     "threaded/scheduler.h",
                     "threaded/scheduler_instance.h",
                     "threaded/scheduler_sync_tag_advance.c",
                     "threaded/scheduler_" + scheduler + ".c",
```

Comment:
This line seems to not be necessary for the `adaptive` scheduler since `scheduler.getRelativePaths()` already includes this. Would it make sense to have all these files added in the enum constructor above for all schedulers?

Reply:
Oh, good catch... I'll fix that. The other three files -- `scheduler.h`, `scheduler_instance.h`, and `scheduler_sync_tag_advance.c` -- are currently used by all schedulers, I think, so to me it seems reasonable to limit redundancy by keeping them here.

(BTW, I have for a while been making noises about factoring this logic out into CMake files, so if others agree with me then this code might not be here for long anyway.)

Reply:
> Factoring this logic out into CMake files

This sounds like a great idea. Are you imagining each scheduler having a separate `CMakeLists.txt` or one `CMakeLists.txt` for all the schedulers?

One thing to keep in mind is that we should still be able to accommodate `cmake: false` (i.e., building without CMake).

Reply:
How important is it to accommodate `cmake: false`? Maybe this isn't the place to have this discussion, but I am not aware of use-cases for `cmake: false`, and what I was talking about was basically to do away with it. Do we maintain it just for fun?

Reply:
It’s for embedded systems, particularly  bare-iron, that don’t necessarily have a supported CMake-based workflow.